### ISSUE TITLE
Report Generator Broken in 4.5.1

### ISSUE THREAD

**Issue by tomdotorg** [poblabs/weewx-belchertown#662]
Please don't delete these questions! They are asked almost with every issue - answering them as below saves us time from asking them!

**First step**
Try to debug the skin yourself. If you can't figure it out, please include the debug information (text, screenshots, etc.). See this guide on debugging the skin. https://github.com/poblabs/weewx-belchertown#how-to-use-debug

**Describe the bug**

On generation, the following error occurs on 4.5.1. The exact same config works fine in 4.4.0.

root@weewx-deployment-vt-dev-6b97859579-lrv5m:~# ./bin/wee_reports
Using configuration file /home/weewx/weewx.conf
Generating for all time
Traceback (most recent call last):
  File "/home/weewx/bin/user/belchertown.py", line 937, in get_extension_list
    response = urlopen( req )
  File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/weewx/bin/weewx/reportengine.py", line 196, in run
    obj.start()
  File "/home/weewx/bin/weewx/reportengine.py", line 281, in start
    self.run()
  File "/home/weewx/bin/weewx/cheetahgenerator.py", line 152, in run
    ngen = self.generate(gen_dict[section_name], self.gen_ts)
  File "/home/weewx/bin/weewx/cheetahgenerator.py", line 222, in generate
    ngen += self.generate(section[subsection], gen_ts)
  File "/home/weewx/bin/weewx/cheetahgenerator.py", line 222, in generate
    ngen += self.generate(section[subsection], gen_ts)
  File "/home/weewx/bin/weewx/cheetahgenerator.py", line 311, in generate
    default_binding)
  File "/home/weewx/bin/weewx/cheetahgenerator.py", line 387, in _getSearchList
    searchList += obj.get_extension_list(timespan, db_lookup)
  File "/home/weewx/bin/user/belchertown.py", line 971, in get_extension_list
    raise Warning( "Error downloading forecast data. Check the URL in your configuration and try again. You are trying to use URL: %s, and the error is: %s" % ( forecast_url, error ) )
Warning: Error downloading forecast data. Check the URL in your configuration and try again. You are trying to use URL: https://api.aerisapi.com/forecasts/44.491,-71.689?&format=json&filter=day&limit=7&client_id=sb0XrM2COVBPB4UcVLszq&client_secret=RKNWqQk04PXWsAMY1a71DVXEZhvbMvCvyB26Zc63, and the error is: HTTP Error 400: Bad Request
Killed

**Link to your website (IMPORTANT)**

weewx-vt-dev.tom.org

**Version of the skin you're using**

1.2

**To Reproduce**

report generator fails every time.

**Expected behavior**

report generator runs successfully as it does in 4.4.0

**Screenshots**

**Device Information (please complete the following information):**
 - Device: [e.g. desktop, iPhone6]
 - OS: [e.g. Windows, Android, iOS]
 - Browser [e.g. Chrome, Safari]

server side. 

**Additional context**
Add any other context about the problem here. Maybe a potential fix?

**HELP WANTED**
If you have a fix for this, please submit a pull request against the development branch!

**Comment by Topslakr** [comment]
@tomdotorg  - I'm running 4.5.1 without any issue. My charts continue to generate normally.

I had to update a couple of WeeWx files though, to replace 'python' in the header line with python3, but that wasn't the fault of Belchertown.

I would take a peek around your files to make sure the system is trying to run everything with the correct version of Python?

**Comment by tomdotorg** [comment]
Hmm. The version is 3.6.9 for both 4.4 and 4.5.1. When you say to look around my files, which ones are you thinking? I run these in Docker containers so it is easy to switch back and forth with the same configuration.

If this is my problem with downloading the Aeris forecast, I guess I will have to figure it out. I just thought it was interesting that it works in 4.4 and not in 4.5.1.

Thanks for the response.

**Comment by Topslakr** [comment]
When I first boot the system, I do run wee_report once (Which throws the Aeris forecast error) but the json file is downloaded as part of that process and then it all runs normally. I have it setup as a cron job, but in a container you may just need to get through two archive intervals to clear that up...

That being said, I do update the top of 'weewxd' to say python3, which was a show stopper when I updated to 4.5.1. And, the setup.py file, to install/update weewx, needs similar before it runs properly.

**Comment by tomdotorg** [comment]
I have had the forecast download problem that rectified after a couple of archive intervals (report cycles). This one is different. I actually discovered it by accident, after seeing that my Belchertown HTML did not update for several days. It cycled through the error I started this bug with every time. It did not fix itself after a few intervals.

I checked the logs, and I am running Python3.

**Comment by michaelundwd** [comment]
For some reason I could not fathom when I was adding the 1hr and 3hr forecasts, it takes weewx a couple of archive loops to sort itself out.
I used the Aeris api call from the last line of your syslog and it returned "the client provided is invalid".  I tried again, using my own id and pwd and it downloaded the forecast ok.  Suggest you do the same and then check the client id and pwd being used, unless of course you changed them in the copy of the log you enclosed.
The syslog should be showing which forecast has failed, but it looks as if it is still reporting the 24hr one.  I must fix that.
